x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -62,6 +81,13 @@ export const scoreApp = (term: string, appLink: AppLink): number => {
   return 0;
 };
 
+const scoreAppByKeywords = (term: string, keywords: string[]): number => {
+  const scores = keywords.map((keyword) => {
+    return scoreAppByTerms(term, keyword);
```

Comment:
Implements the same algorithm used for scoring by title under the assumption that applications would only use a limited number of keywords. We don't restrict the number of keywords that can be added though.

Reply:
Even with 1000 keywords, I can't imagine this being slow, but I don't know what the implementation of the levenstein distance package we're using looks like

Reply:
> If we are given two strings of size n1 and n2, the standard Levenshtein edit distance computation is by a dynamic algorithm with time complexity O(n1n2) and space complexity O(n1n2)

I think this is alright. OTOH, the more I think about it, the more I think I may have over-engineered the scoring algorithm. Maybe exact+containing matches are good enough. After all, our SO provider (well, the used tokenizer/analyzer) only returns exact + starting-by results. Should we just remove the lev distance matching?

WDYT?

Reply:
@pgayvallet I can't imaging that plugins will register more than a couple, maybe a few at most, keywords/aliases because they have to be careful that they don't register the same keywords as other plugins have. 
That being said, maybe we can keep it as it is right now and if we see a drastic perf issue down the line, we could remove it or at least remove it from scoring against keywords?

Reply:
No, keeping it is totally fine imho. My remark was more about the functional behavior than performance.